src/tactic/core.lean

```diff
@@ -2114,3 +2114,12 @@ add_tactic_doc
   tags                     := ["simplification"] }
 
 end tactic
+
+/--
+`find_defeq red m e` looks for a key in `m` that is defeq to `e` (up to transparency `red`),
+and returns the value associated with this key if it exists.
+Otherwise, it fails.
+-/
+meta def list.find_defeq (red : tactic.transparency) {v} (m : list (expr × v)) (e : expr) :
```

Comment:
I wasn't really sure where to put this function.

Reply:
I don't know either.  We need to clean up `tactic.core` at some point anyhow.  It's over *two thousand* lines of unrelated code.